src/NavItem.js

```diff
@@ -23,32 +26,37 @@ const NavItem = React.createClass({
 
   render() {
     let {
+        role,
+        linkId,
         disabled,
         active,
         href,
         title,
         target,
         children,
+        'aria-controls': ariaControls,  // eslint-disable-line react/prop-types
         ...props } = this.props; // eslint-disable-line object-shorthand
     let classes = {
           active,
           disabled
         };
     let linkProps = {
+          role,
           href,
           title,
           target,
+          id: linkId,
           onClick: this.handleClick,
           ref: 'anchor'
         };
 
-    if (href === '#') {
+    if (!role && href === '#') {
       linkProps.role = 'button';
     }
 
     return (
-      <li {...props} className={classNames(props.className, classes)}>
-        <a {...linkProps}>
+      <li {...props} role='presentation' className={classNames(props.className, classes)}>
+        <a {...linkProps} aria-selected={active} aria-controls={ariaControls}>
```

Comment:
Isn't it supposed to be as `ariaSelected` and `ariaControls` ?
https://facebook.github.io/react/docs/tags-and-attributes.html#supported-attributes
> Note:
All attributes are camel-cased